stan/math/rev/fun/matrix_exp_multiply.hpp

```diff
@@ -14,17 +14,17 @@ namespace math {
 /**
  * Wrapper of matrix_exp_action function for a more literal name
  *
- * @tparam Ta scalar type matrix A
- * @tparam Tb scalar type matrix B
- * @tparam Cb Columns matrix B
+ * @tparam Ta type of the matrix A
+ * @tparam Tb type of the matrix B
  *
  * @param[in] A Matrix
  * @param[in] B Matrix
  * @return exponential of A multiplies B
  */
-template <typename Ta, typename Tb, int Cb>
-inline Eigen::Matrix<return_type_t<Ta, Tb>, -1, Cb> matrix_exp_multiply(
-    const Eigen::Matrix<Ta, -1, -1>& A, const Eigen::Matrix<Tb, -1, Cb>& B) {
+template <typename Ta, typename Tb, require_all_eigen_t<Ta, Tb>* = nullptr,
+          require_any_st_autodiff<Ta, Tb>* = nullptr>
+inline Eigen::Matrix<return_type_t<Ta, Tb>, -1, Tb::ColsAtCompileTime>
```

Comment:
Will this always be guranteed to be dynamic in rows? Or should the return just be `auto`?

Reply:
We can think about auto after we can return expressions. Also I think auto won't work with early return.